web/htdocs/assets/js/codesearch_ui.js

```diff
@@ -49,6 +49,16 @@ var Match = Backbone.Model.extend({
   initialize: function() {
   },
   url: function() {
+    if (this.get('tree') in CodesearchUI.intervalViewRepos) {
+      return this.internalUrl();
+    } else {
+      return this.externalUrl();
+    }
+  },
+  internalUrl: function() {
+    return "/view/" + this.get('tree') + "/" + this.get('path') + "#L" + this.get('lno');
```

Comment:
fwiw there's one affordance for handling this right now, which is the `-revparse` flag to `codesearch`, which causes it to resolve all commit references into sha1's while indexing, which will then propagate through the system to ensure stable references and links.

I'm a big fan of linking to sha1's in general, but there are definitely a bunch of fiddly UX questions.